ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.java

```diff
@@ -346,25 +346,25 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
       return false;
     }
 
+    Optional<LocationOfInterest> loi = Optional.ofNullable(state.getLocationOfInterest());
     if (item.getItemId() == R.id.move_loi_menu_item) {
       hideBottomSheet();
       mapContainerViewModel.setMode(Mode.MOVE_POINT);
-      mapContainerViewModel.setReposLocationOfInterest(state.getLocationOfInterest());
+      mapContainerViewModel.setReposLocationOfInterest(loi);
       Toast.makeText(getContext(), R.string.move_point_hint, Toast.LENGTH_SHORT).show();
     } else if (item.getItemId() == R.id.delete_loi_menu_item) {
-      Optional<LocationOfInterest> locationOfInterestToDelete = state.getLocationOfInterest();
-      if (locationOfInterestToDelete.isPresent()) {
+      if (loi.isPresent()) {
         new Builder(requireActivity())
             .setTitle(
                 getString(
                     R.string.loi_delete_confirmation_dialog_title,
-                    locationOfInterestHelper.getLabel(locationOfInterestToDelete)))
+                    locationOfInterestHelper.getLabel(loi)))
             .setMessage(R.string.loi_delete_confirmation_dialog_message)
             .setPositiveButton(
                 R.string.delete_button_label,
                 (dialog, id) -> {
                   hideBottomSheet();
-                  viewModel.deleteLocationOfInterest(locationOfInterestToDelete.get());
+                  viewModel.deleteLocationOfInterest(loi.get());
```

Comment:
Why is the LOI being deleted? What does that actually do?

Reply:
Earlier, the app provided the capability to remove the feature if you're the owner of the project or if the feature was added by you. This should be removed along with the ui components.